net/ipv4/tcp_output.c

```diff
@@ -2404,6 +2476,7 @@ static int tcp_mtu_probe(struct sock *sk)
 	mss_now = tcp_current_mss(sk);
 	probe_size = tcp_mtu_to_mss(sk, (icsk->icsk_mtup.search_high +
 				    icsk->icsk_mtup.search_low) >> 1);
+	TFW_ADJUST_TLS_OVERHEAD(probe_size);
```

Comment:
Maybe we should adjust `probe_size` only after all necessary checks right before this expression?
```
if (!tcp_can_coalesce_send_queue_head(sk, probe_size))
		return -1;
```

Otherwise it looks like, reducing the size(TFW_ADJUST_TLS_OVERHEAD), checks window `if (tp->snd_wnd < size_needed)`, and during encryption we add `TLS_OVERHEAD`, after that total size might be greater that `tp->snd_wnd`.

Reply:
yes fixed